File: js/Strategy.js
var PositionStrategy = {
  tier_one: ["AA", "KK", "QQ", "JJ", "AK"],
  tier_two: ["TT", "99", "88", "77", "AQ", "AJ"],
  tier_three: ["ATs", "KQs", "QJs", "JTs"],
  tier_four: ["66", "55", "44", "33", "22","A9s", "109s", "98s"],
  };

var position_dictionary_nine = {
        1: "Small Blind",
        2: "Early Position",
        3: "Early Position",
        4: "Middle Position",
        5: "Middle Position",
        6: "Middle Position",
        7: "Late Position",
        8: "Late Position",
        9: "Button",
        };

function printDecision(decision) {
  var placeholder = document.getElementById('Decision');
  placeholder.innerHTML = decision;
  }

function getAction(cards, position) {
  if (position_dictionary_nine[position.value] == "Small Blind"){
    if (PositionStrategy.tier_one.includes(cards.value)){
      return "Bet";
      } else {
        return "Fold";
      }
    } else {
      return "No position dict";
    }
  }

var button = document.getElementById('cardSubmit');

button.onclick = function(){
    var cards = document.getElementById("cards");
    var position = document.getElementById("position");
    var decision = getAction(cards, position);
    printDecision(decision);
  }

File: Strategy.py
class PositionStrategy(object):
    TIER_ONE = ["AA", "KK", "QQ", "JJ", "AK", "AKs"]
    TIER_TWO = ["TT", "99", "88", "77", "AQ", "AJ", "AQs", "AJs"]
    TIER_THREE = ["ATs", "KQs", "QJs", "JTs"]
    TIER_FOUR = ["66", "55", "44", "33", "22","A9s", "109s", "98s"]

    def __init__(self, Tournament):
        self.tournament_position_dict = Tournament.create_position_dictionary()

    def action(self, cards, position):
        self.cards = cards
        if self.tournament_position_dict[position] == "Small Blind":
            self.small_blind_action()
        elif self.tournament_position_dict[position] == "Early Position":
            self.early_position_action()
        elif self.tournament_position_dict[position] == "Middle Position":
            self.middle_position_action()
        elif self.tournament_position_dict[position] == "Late Position":
            self.late_position_action()
        elif self.tournament_position_dict[position] == "Button":
            self.button_action()

    def small_blind_action(self):
        if self.cards in PositionStrategy.TIER_ONE:
            print("Raise limper or re-raise any standard bet.")
        elif self.cards in PositionStrategy.TIER_TWO:
            print("Raise limper or call any standard bet.")
        else:
            print("Call in unraised pot, else fold.")

    def early_position_action(self):
        if self.cards in PositionStrategy.TIER_ONE:
            print("Raise limper or re-raise any standard bet.")
        elif self.cards in PositionStrategy.TIER_TWO:
            print("Raise limper or call any standard bet.")
        else:
            print("Fold")

    def middle_position_action(self):
        if self.cards in PositionStrategy.TIER_ONE:
            print("Raise if first in, Else raise limper or reraise.")
        elif self.cards in PositionStrategy.TIER_TWO:
            print("Raise if first in, Else call standard raise.")
        elif self.cards in PositionStrategy.TIER_THREE:
            print("Raise if first in, Else call standard raise.")
        else:
            print("Fold")

    def late_position_action(self):
        print("If first in raise,")
        if self.cards in PositionStrategy.TIER_ONE:
            print("Else raise limper or reraise.")
        elif self.cards in PositionStrategy.TIER_TWO:
            print("Else call standard raise.")
        elif self.cards in PositionStrategy.TIER_THREE:
            print("Else call standard raise.")
        elif self.cards in PositionStrategy.TIER_FOUR:
            print("Else call standard raise.")
        else:
            print("Else fold.")

    def button_action(self):
        print("If first in raise,")
        if self.cards in PositionStrategy.TIER_ONE:
            print("Else raise limper or reraise.")
        elif self.cards in PositionStrategy.TIER_TWO:
            print("Else call standard raise.")
        elif self.cards in PositionStrategy.TIER_THREE:
            print("Else call standard raise.")
        elif self.cards in PositionStrategy.TIER_FOUR:
            print("Else call standard raise.")
        else:
            print("Else limp in or call standard raise with any two cards.")
